lib/auth/github.go

```diff
@@ -952,7 +956,26 @@ func ValidateClientRedirect(clientRedirect string, ssoTestFlow bool, settings *t
 	case "http":
 		switch u.Hostname() {
 		default:
-			return trace.BadParameter("invalid hostname in client redirect URL")
+			if settings == nil || len(settings.InsecureAllowedCidrRanges) == 0 {
+				return trace.BadParameter("invalid hostname in client redirect URL")
+			}
+			hostIP, err := netip.ParseAddr(u.Hostname())
+			if err != nil {
+				return trace.BadParameter("custom client HTTP redirect URLs must be IP addresses")
+			}
+
+			for _, cidrStr := range settings.InsecureAllowedCidrRanges {
+				cidr, err := netip.ParsePrefix(cidrStr)
+				if err != nil {
+					return trace.Wrap(err)
+				}
+				if cidr.Contains(hostIP) {
+					return nil
+				}
+			}
+
+			return trace.AccessDenied("unknown custom client redirect URL hostname")
```

Comment:
Do we have to make a distinction between the auth connector not allowing any cidr ranges and the hostname not matching any?

Reply:
N/A in the current simplified code